persistence-migration/src/main/java/org/dependencytrack/persistence/migration/change/v550/ComputeSeveritiesChange.java

```diff
@@ -0,0 +1,244 @@
+        } else if (cvssSeverity != null) {
+            severity = cvssSeverity;
+        }
+
+        return severity;
+    }
+
+    private static Severity normalizedCvssV2Score(final double score) {
+        if (score >= 7) {
+            return Severity.HIGH;
+        } else if (score >= 4) {
+            return Severity.MEDIUM;
+        } else if (score > 0) {
+            return Severity.LOW;
+        } else {
+            return Severity.UNASSIGNED;
+        }
+    }
+
+    private static Severity normalizedCvssV3Score(final double score) {
+        if (score >= 9) {
+            return Severity.CRITICAL;
+        } else if (score >= 7) {
+            return Severity.HIGH;
+        } else if (score >= 4) {
+            return Severity.MEDIUM;
+        } else if (score > 0) {
+            return Severity.LOW;
+        } else {
+            return Severity.UNASSIGNED;
+        }
+    }
+
+    private static final Map<Severity, Map<Severity, Severity>> OWASP_RR_LIKELIHOOD_TO_IMPACT_SEVERITY_MATRIX = Map.of(
+            Severity.LOW, Map.of(
+                    Severity.LOW, Severity.INFO,
+                    Severity.MEDIUM, Severity.LOW,
+                    Severity.HIGH, Severity.MEDIUM,
+                    Severity.UNASSIGNED, Severity.UNASSIGNED
+            ),
+            Severity.MEDIUM, Map.of(
+                    Severity.LOW, Severity.LOW,
+                    Severity.MEDIUM, Severity.MEDIUM,
+                    Severity.HIGH, Severity.HIGH,
+                    Severity.UNASSIGNED, Severity.UNASSIGNED
+            ),
+            Severity.HIGH, Map.of(
+                    Severity.LOW, Severity.MEDIUM,
+                    Severity.MEDIUM, Severity.HIGH,
+                    Severity.HIGH, Severity.CRITICAL,
+                    Severity.UNASSIGNED, Severity.UNASSIGNED
+            ),
+            Severity.UNASSIGNED, Map.of(
+                    Severity.LOW, Severity.UNASSIGNED,
+                    Severity.MEDIUM, Severity.UNASSIGNED,
+                    Severity.HIGH, Severity.UNASSIGNED,
+                    Severity.UNASSIGNED, Severity.UNASSIGNED
+            )
+    );
+
+    private static Severity normalizedOwaspRRScore(final double likelihoodScore, final double technicalImpactScore, final double businessImpactScore) {
+        double impactScore = Math.max(technicalImpactScore, businessImpactScore);
+        Severity likelihoodSeverity = normalizedOwaspRRScore(likelihoodScore);
+        Severity impactSeverity = normalizedOwaspRRScore(impactScore);
+        return OWASP_RR_LIKELIHOOD_TO_IMPACT_SEVERITY_MATRIX.get(likelihoodSeverity).get(impactSeverity);
+    }
+
+    private static Severity normalizedOwaspRRScore(final double score) {
+        if (score >= 6) {
+            return Severity.HIGH;
+        } else if (score >= 3) {
+            return Severity.MEDIUM;
+        } else if (score > 0) {
+            return Severity.LOW;
+        } else {
+            return Severity.UNASSIGNED;
+        }
+    }
```

Comment:
Copying this code was necessary to avoid a circular dependency between `apiserver` and `persistence-migration`.